### DISCUSSION TITLE
MapInfo struct does not contain BTF information

### DISCUSSION THREAD

**Discussion by SirNexus** [cilium/ebpf#1049] [Q&A]
The [mapInfo](https://github.com/cilium/ebpf/blob/73ed4e8864580e771e00be8771d45a159da3617b/info.go#L24) struct does not contain BTF information for the map its representing. However this information is accessible in the underlying [sys/MapInfo](https://github.com/cilium/ebpf/blob/73ed4e8864580e771e00be8771d45a159da3617b/internal/sys/types.go#L494) struct.

The BTF information (BtfId, BtfVmlinuxValueTypeId) should be exposed to power use cases such as map dumping similar to how bpftool does it without having to locate the associated program that loaded it.

Is there a reason this information is not exposed?

**Comment by lmb** [comment]
The reason is that nobody had a use for it so far. Feel free to submit a PR if you want to add it.

**Comment by ti-mo** [comment]
> without having to locate the associated program that loaded it

This won't help on programs loaded by ebpf-go itself, as we only marshal and load the types needed to construct func_infos.

To replicate bpftool behaviour, you need the type info for arbitrary maps loaded by pin/id, correct? It would be nice to expose this through Map.Key() and .Value(), much like we do for MapSpec.
